### DISCUSSION TITLE
volt

### DISCUSSION THREAD

**Discussion by jdion84** [spatie/laravel-honeypot#122] [Q&A]
will this package become compatible with livewire volt in near future or nah?

**Comment by freekmurze** [comment]
Feel free to submit a PR with the desired changes or nah 🙂

**Comment by jdion84** [comment]
there is an api to do this but this package requires the property to be typed

for example, we could do:

```php
use Spatie\Honeypot\Http\Livewire\Concerns\UsesSpamProtection;
use Spatie\Honeypot\Http\Livewire\Concerns\HoneypotData;
use function Livewire\Volt\{uses, state};

uses(UsesSpamProtection::class);

state([
    'extraFields' => new HoneypotData,
]);

$register = function () {
    $this->protectAgainstSpam();

    //
};
```

however this throws an error because we have not explicitly typed the `extraFields` property like we would in a class. if you're aware of a workaround then this package would not need any changes to work with volt, and I can submit PR for readme updates

**Comment by jcsoriano** [comment]
Hi @jdion84 and @freekmurze , created a PR to update the docs with instructions on how to use the package with Volt in functional syntax :) https://github.com/spatie/laravel-honeypot/pull/125
